### PR TITLE
Update proposal.md using pyquil quilt-demo branch changes

### PR DESCRIPTION
**Pull request by notmgsk**
TODO:
- [x] Include the updated `Quil.g4`
- [x] Get feedback

### CODE REVIEW

File: rfcs/analog/proposal.md [hunk 1]
```diff
@@ -65,14 +63,13 @@ that can be modified throughout the program using SET-* instructions (and
...
-| Frequency | (not set)     | Real numbers          | No                    |
-| Phase     | 0.0           | Real numbers          | Yes                   |
-| Scale     | 1.0           | Real numbers          | No                    |
+tracked through the program:
 
+| Name      | Initial Value                     | Valid Values          | Can be parameterized? |
+|-----------|-----------------------------------|-----------------------|-----------------------|
+| Frequency | INITIAL-FREQUENCY in DEFFRAME     | Real numbers          | Yes                   |
+| Phase     | 0.0                               | Real numbers          | Yes                   |
+| Scale     | 1.0                               | Real numbers          | Yes                   |
```
Comment (notmgsk) [inline rfcs/analog/proposal.md:72]: `Frequency` and `Scale` are now marked as `Yes` under `Can be parameterized?`. Are there examples of this in the proposal?

File: rfcs/analog/proposal.md [hunk 2]
```diff
@@ -102,11 +99,11 @@ as an integration kernel. The inner product of the integration kernel and the
 list of measured IQ values is evaluated to produce a single complex result.
 
 This complex number needs to be stored in Quil classical memory. Quil does not
-currently support complex numbers, so a real array of length 2 is used instead:
+currently support complex typed memory regions, so a real array of length 2 is used instead:
 ```
 # Simple capture of an IQ point
 DECLARE iq REAL[2]
-CAPTURE 0 "ro" flat(duration: 1e-6, iq: 2+3i) iq
+CAPTURE 0 "ro" boxcar_kernel(duration: 1e-6) iq
```
Comment (notmgsk) [inline rfcs/analog/proposal.md:106]: Why was this changed?

File: rfcs/analog/proposal.md [hunk 3]
```diff
@@ -127,6 +124,8 @@ Roughly speaking:
 A more precise specification of the timing semantics is provided in
 [scheduling.md](./scheduling.md).
 
+TODO Erik?
```
Comment (notmgsk) [inline rfcs/analog/proposal.md:127]: In the pyquil version of the proposal, the above `scheduling.md` link was replaced by a `TODO` comment. Was there more to do? @kilimanjaro

File: rfcs/analog/proposal.md [hunk 4]
```diff
@@ -193,9 +192,11 @@ Calibrations can be parameterized and include concrete values, which are
 resolved in "Haskell-style", with later definitions being prioritized over
 earlier ones. For example, given the following list of calibration definitions
 in this order:
-1. `DEFCAL RX(%theta) %qubit:`
-2. `DEFCAL RX(%theta) 0:`
-3. `DEFCAL RX(pi/2) 0:`
+
+1. `DEFCAL RX(%theta) qubit: ...`
```
Comment (notmgsk) [inline rfcs/analog/proposal.md:196]: I believe `qubit` is the correct syntax over `%qubit`.

File: rfcs/analog/proposal.md [hunk 5]
```diff
@@ -266,6 +267,8 @@ to pulses by the compiler, or they can resolve the pulses themselves
 
 Here are some example calibrations for various types of gates and measurements.
 
+Here are some example calibrations for various types of gates and measurements.
```
Comment (notmgsk) [inline rfcs/analog/proposal.md:270]: Oops. This should not be duplicated.

File: rfcs/analog/proposal.md [hunk 6]
```diff
@@ -374,6 +364,19 @@ SET-SCALE 0 "cz" 0.45
 PULSE 0 1 "cz" erfsquare(duration: 100e-9, risetime: 20e-9, padleft: 0, padright: 0)
 ```
 
+Active Reset Calibration:
```
Comment (notmgsk) [inline rfcs/analog/proposal.md:367]: Was this intentionally removed from the proposal (in the pyquil version)? @kilimanjaro

File: rfcs/analog/proposal.md [hunk 7]
```diff
@@ -374,6 +364,19 @@ SET-SCALE 0 "cz" 0.45
...
+    MEASURE qubit ro[0]
+    JUMP-UNLESS ro @delay
+    RX(pi) qubit
+    JUMP @end
+    LABEL delay
+    DELAY qubit 60e-9
+    LABEL end
+```
+
 ## FAQs
```
Comment (notmgsk) [inline rfcs/analog/proposal.md:380]: The FAQ was not present in the pyquil document. Was that intentional @kilimanjaro?

**Review by notmgsk — COMMENTED** [review]
The `NONBLOCKING` section could maybe be improved. It leads by saying that in some cases you may want to allow concurrent/simultaneous pulses on the same qubit and that this can be achieved with `NONBLOCKING`. But then finishes with saying that two consecutive `NONBLOCKING PULSE 0 "xy" ...` would require serial execution. It is mentioned between these two statements that the `NONBLOCKING` modifier does not exclude pulse operations on *other frames* but I think that should emphasised immediately and repeatedly

File: rfcs/analog/proposal.md (hunk 3 above)
Comment (kalzoo) [reply rfcs/analog/proposal.md:127]: @kilimanjaro and I had talked about this TODO and his suggestion was to remove the reference to the timing semantics entirely, because that spec is not settled yet. Pending his suggestions, I think the best course may be to just make it clear that it is only a proposal, but still link it. Perhaps:

> A more precise specification of the timing semantics has been proposed in ...

then we can remove the TODO.

File: rfcs/analog/proposal.md [hunk 8]
```diff
@@ -65,14 +63,13 @@ that can be modified throughout the program using SET-* instructions (and
...
-| Name      | Initial Value | Valid Values          | Can be parameterized? |
-|-----------|---------------|-----------------------|-----------------------|
-| Frequency | (not set)     | Real numbers          | No                    |
-| Phase     | 0.0           | Real numbers          | Yes                   |
-| Scale     | 1.0           | Real numbers          | No                    |
+tracked through the program:
 
+| Name      | Initial Value                     | Valid Values          | Can be parameterized? |
+|-----------|-----------------------------------|-----------------------|-----------------------|
+| Frequency | INITIAL-FREQUENCY in DEFFRAME     | Real numbers          | Yes                   |
```
[the remark this answers is not in the document]
Comment (kalzoo) [reply rfcs/analog/proposal.md:70]: Do you mean [this](https://github.com/rigetti/quil/blob/bugfix/quilt-rfc-improvements/rfcs/analog/spec_changes.md#standard-frame-attributes)?

Also, shortly above that, this:

> The specific set of required and optional frame attributes is implementation specific.

File: rfcs/analog/proposal.md (hunk 2 above)
Comment (kalzoo) [reply rfcs/analog/proposal.md:106]: Great question, esp as IIRC `boxcar_kernel` is not in fact supported

File: rfcs/analog/proposal.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (notmgsk) [reply rfcs/analog/proposal.md:70]: Yeah, that'll do. Cheers

**Comment by notmgsk** [comment]
@kalzoo Shall we merge this, and move over any comments into the main PR?

**Comment by kalzoo** [comment]
> @kalzoo Shall we merge this, and move over any comments into the main PR? Then onto addressing #45 and merging in `spec_changes.md`.

Sure, sounds good
